azure/blob/fs.sh

```diff
@@ -0,0 +1,202 @@
+# unable to create any of the resources.
+# This is specially useful on read/write scenarios.
+#
+# If it succeeds you can use the returned instance to list/send/get files from
+# Azure BLOB storage.
+fn azure_blob_fs_create(resgroup, location, accountname, sku, tier, containername) {
+	# WHY: for now creating a group that already exists is ok
+	# and the function does not return any value, it will abort =/
+	# changing it now may break other people.
+	azure_group_create($resgroup, $location)
+
+	status <= azure_storage_account_exists($accountname, $resgroup)
+	if $status != "0" {
+		err <=  azure_storage_account_create_blob(
+		    $accountname,
+		    $resgroup,
+		    $location,
+		    $sku,
+		    $tier
+		)
+		if $err != "" {
+			return (), $err
+		}
+        }
+
+        # WHY: creating a container that already exists do not fail on az
+        err <= azure_storage_container_create_by_resgroup(
+                $containername,
+                $accountname,
+                $resgroup,
+        )
+        if $err != "" {
+		return (), $err
+        }
+	return azure_blob_fs_new($resgroup, $accountname, $containername), ""
+}
+
+# Creates a new blob fs instance, just like azure_blob_fs_create but
+# it will not attempt to create any resource. Useful for reading operations.
+fn azure_blob_fs_new(resgroup, accountname, containername) {
+	return ($resgroup $accountname $containername)
+}
+
+# Uploads a single file
+fn azure_blob_fs_upload(fs, remotepath, localpath) {
+	return azure_storage_blob_upload_by_resgroup(
+		azure_blob_fs_container($fs),
+		azure_blob_fs_account($fs),
+		azure_blob_fs_resgroup($fs),
+		$remotepath,
+		$localpath
+	)
+}
+
+# Uploads a local dir to a remote dir.
+# It will not create the local dir on the remote dir, only its
+# contents (including other dirs, recursively).
+#
+# For example, copying the local dir /klb to the remote dir /test
+# will copy all contents of /klb to /test, like /test/file1 and /test/dir1/file1
+# but it will not create a "klb" directory inside the remote dir.
+#
+# This is the behavior of Plan9 dircp (http://man.cat-v.org/plan_9/1/tar)
+# and it seems more intuitive than the cp -r behavior of creating only the
+# base dir of the source path at the target path if it does not exists
+# and to copy only the contents you need the hack of regex expansion:
+# cp -r /srcdir/* /targetdir.
+#
+# This function returns an error string if it fails and "" if it succeeds.
+fn azure_blob_fs_upload_dir(fs, remotedir, localdir) {
+	# WHY: Make code handling results uniform (no relative path handling)
+	localdir <= realpath $localdir
+	out, status <= tree -if --noreport $localdir
+	if $status != "0" {
+		return format("error listing directory[%s], output: %s", $localpath, $out)
+	}
+	all <= split($out, "\n")
+	files = ()
+	for a in $all {
+		_, status <= test -f $a
+		if $status == "0" {
+			files <= append($files, $a)
+		}
+	}
+
+	filesprefix <= format("s:^%s::", $localdir)
```

Comment:
this line bugged me for a while... I thought it was some Go obscure syntax haha
Maybe rename the variable to `prefixRegex` or something related to `pattern`?